raft/src/main/java/org/apache/kafka/raft/RaftClient.java

```diff
@@ -172,15 +176,17 @@ default void beginShutdown() {}
      * uncommitted entries after observing an epoch change.
      *
      * @param epoch the current leader epoch
+     * @param requiredEndOffset if this is set, it is the offset we must use as the end offset (inclusive).
```

Comment:
This is the same base or starting offset. The `RaftClient` does expose LEO but it does expose a base offset. I think we should do something like this:
```java
     * @param expectedBaseOffset if this is set, it matches the base offset that KRaft will use for the {@code records}.
```

Reply:
ok. we can use `requiredBaseOffset`.